### ISSUE TITLE
[IDF 5.0] esp_partition_mmap failed on esp32s3 with big partition(12MB) when psram is enabled. (IDFGH-8372)

### ISSUE THREAD

**Issue by inapp123** [espressif/esp-idf#9841]
### Answers checklist.

- [X] I have read the documentation [ESP-IDF Programming Guide](https://docs.espressif.com/projects/esp-idf/en/latest/) and the issue is not addressed there.
- [X] I have updated my IDF branch (master or release) to the latest version and checked that the issue is present there.
- [X] I have searched the issue tracker for a similar issue and not found a similar issue.

### IDF version.

v5.0-beta1-641-gc321739074

### Operating System used.

Windows

### How did you build your project?

Command line with idf.py

### If you are using Windows, please specify command line type.

PowerShell

### Development Kit.

Custom board with ESP32S3-WROOM-1-N16R8 module

### Power Supply used.

USB

### What is the expected behavior?

Partition being mapped successfully

### What is the actual behavior?

It crash with ESP_ERR_NO_MEM

### Steps to reproduce.

Code can be found [here](https://github.com/inapp123/esp32s3-mmap-test)

1. use `idf.py menuconfig` to enable psram (Octal psram) and change serial flasher - flash size to 16M to match module specification.
2. config partition table to add a new custom partition with size 12MB, labeled "asset".
3. build and flash the code.

the main part of code is:
```c
    const esp_partition_t* part = esp_partition_find_first(0x40, 0x00, "asset");
    if (part == NULL) {
        ESP_LOGE(TAG, "Failed to find asset partition");
        abort();
    }    
    spi_flash_mmap_dump();
    ESP_LOGI(TAG, "Found asset partition at offset 0x%" PRIx32 " with size %" PRIx32, part->address,part->size);


    uint8_t* mmaped_ptr;
    spi_flash_mmap_handle_t handle;
    ESP_ERROR_CHECK(esp_partition_mmap(part, 0, part->size, SPI_FLASH_MMAP_DATA, &(mmaped_ptr),&handle));
    ESP_LOGI(TAG, "Mounted at offset 0x%" PRIx32 "", (uint32_t)mmaped_ptr);
```

### Debug Logs.

```plain
I (29) boot: ESP-IDF v5.0-beta1-641-gc321739074 2nd stage bootloader
I (29) boot: compile time 23:59:38
I (29) boot: chip revision: V001
I (33) boot_comm: chip revision: 1, min. bootloader chip revision: 0
I (40) boot.esp32s3: Boot SPI Speed : 80MHz
I (45) boot.esp32s3: SPI Mode       : DIO
I (49) boot.esp32s3: SPI Flash Size : 16MB
I (54) boot: Enabling RNG early entropy source...
I (60) boot: Partition Table:
I (63) boot: ## Label            Usage          Type ST Offset   Length
I (70) boot:  0 otadata          OTA data         01 00 00009000 00002000
I (78) boot:  1 phy_init         RF data          01 01 0000b000 00001000
I (85) boot:  2 factory          factory app      00 00 00010000 00100000
I (93) boot:  3 ota_0            OTA app          00 10 00110000 00100000
I (100) boot:  4 ota_1            OTA app          00 11 00210000 00100000
I (108) boot:  5 nvs              WiFi data        01 02 00310000 000e1000
I (115) boot:  6 asset            unknown          40 00 003f1000 00c00000
I (123) boot: End of partition table
I (127) boot: Defaulting to factory image
I (132) boot_comm: chip revision: 1, min. application chip revision: 0
I (139) esp_image: segment 0: paddr=00010020 vaddr=3c020020 size=08a9ch ( 35484) map
I (154) esp_image: segment 1: paddr=00018ac4 vaddr=3fc91e00 size=03134h ( 12596) load
I (159) esp_image: segment 2: paddr=0001bc00 vaddr=40374000 size=04418h ( 17432) load
I (168) esp_image: segment 3: paddr=00020020 vaddr=42000020 size=19708h (104200) map
I (192) esp_image: segment 4: paddr=00039730 vaddr=40378418 size=09904h ( 39172) load
I (201) esp_image: segment 5: paddr=0004303c vaddr=50000000 size=00010h (    16) load
I (207) boot: Loaded app from partition at offset 0x10000
I (207) boot: Disabling RNG early entropy source...
I (221) octal_psram: vendor id    : 0x0d (AP)
I (221) octal_psram: dev id       : 0x02 (generation 3)
I (221) octal_psram: density      : 0x03 (64 Mbit)
I (226) octal_psram: good-die     : 0x01 (Pass)
I (231) octal_psram: Latency      : 0x01 (Fixed)
I (237) octal_psram: VCC          : 0x01 (3V)
I (242) octal_psram: SRF          : 0x01 (Fast Refresh)
I (248) octal_psram: BurstType    : 0x01 (Hybrid Wrap)
I (253) octal_psram: BurstLen     : 0x01 (32 Byte)
I (259) octal_psram: Readlatency  : 0x02 (10 cycles@Fixed)
I (265) octal_psram: DriveStrength: 0x00 (1/1)
I (270) esp_psram: Found 8MB PSRAM device
I (275) esp_psram: Speed: 40MHz
I (279) cpu_start: Pro cpu up.
I (282) cpu_start: Starting app cpu, entry point is 0x403752f0
0x403752f0: call_start_cpu1 at C:/Users/ShiroganeTsumugi/esp/esp-idf/components/esp_system/port/cpu_start.c:142

I (0) cpu_start: App cpu up.
I (1020) esp_psram: SPI SRAM memory test OK
I (1029) cpu_start: Pro cpu start user code
I (1029) cpu_start: cpu freq: 160000000 Hz
I (1030) cpu_start: Application information:
I (1033) cpu_start: Project name:     template-app
I (1038) cpu_start: App version:      1
I (1043) cpu_start: Compile time:     Sep 23 2022 23:59:18
I (1049) cpu_start: ELF file SHA256:  497afe5d0adc4f7b...
I (1055) cpu_start: ESP-IDF:          v5.0-beta1-641-gc321739074
I (1062) heap_init: Initializing. RAM available for dynamic allocation:
I (1069) heap_init: At 3FC959A8 len 00053D68 (335 KiB): D/IRAM
I (1075) heap_init: At 3FCE9710 len 00005724 (21 KiB): STACK/DRAM
I (1082) heap_init: At 3FCF0000 len 00008000 (32 KiB): DRAM
I (1088) heap_init: At 600FE010 len 00001FF0 (7 KiB): RTCRAM
I (1095) esp_psram: Adding pool of 8192K of PSRAM memory to heap allocator
I (1103) spi_flash: detected chip: gd
I (1107) spi_flash: flash io: dio
I (1112) cpu_start: Starting scheduler on PRO CPU.
I (0) cpu_start: Starting scheduler on APP CPU.
I (1131) esp_psram: Reserving pool of 32K of internal memory for DMA/internal allocations
page 0: refcnt=1 paddr=2
page 1: refcnt=1 paddr=3
page 2: refcnt=1 paddr=1
page 3: refcnt=1 paddr=32768
page 4: refcnt=1 paddr=32769
page 5: refcnt=1 paddr=32770
page 6: refcnt=1 paddr=32771
page 7: refcnt=1 paddr=32772
page 8: refcnt=1 paddr=32773
page 9: refcnt=1 paddr=32774
page 10: refcnt=1 paddr=32775
page 11: refcnt=1 paddr=32776
page 12: refcnt=1 paddr=32777
page 13: refcnt=1 paddr=32778
page 14: refcnt=1 paddr=32779
page 15: refcnt=1 paddr=32780
page 16: refcnt=1 paddr=32781
page 17: refcnt=1 paddr=32782
page 18: refcnt=1 paddr=32783
page 19: refcnt=1 paddr=32784
page 20: refcnt=1 paddr=32785
page 21: refcnt=1 paddr=32786
page 22: refcnt=1 paddr=32787
page 23: refcnt=1 paddr=32788
page 24: refcnt=1 paddr=32789
page 25: refcnt=1 paddr=32790
page 26: refcnt=1 paddr=32791
page 27: refcnt=1 paddr=32792
page 28: refcnt=1 paddr=32793
page 29: refcnt=1 paddr=32794
page 30: refcnt=1 paddr=32795
page 31: refcnt=1 paddr=32796
page 32: refcnt=1 paddr=32797
page 33: refcnt=1 paddr=32798
page 34: refcnt=1 paddr=32799
page 35: refcnt=1 paddr=32800
page 36: refcnt=1 paddr=32801
page 37: refcnt=1 paddr=32802
page 38: refcnt=1 paddr=32803
page 39: refcnt=1 paddr=32804
page 40: refcnt=1 paddr=32805
page 41: refcnt=1 paddr=32806
page 42: refcnt=1 paddr=32807
page 43: refcnt=1 paddr=32808
page 44: refcnt=1 paddr=32809
page 45: refcnt=1 paddr=32810
page 46: refcnt=1 paddr=32811
page 47: refcnt=1 paddr=32812
page 48: refcnt=1 paddr=32813
page 49: refcnt=1 paddr=32814
page 50: refcnt=1 paddr=32815
page 51: refcnt=1 paddr=32816
page 52: refcnt=1 paddr=32817
page 53: refcnt=1 paddr=32818
page 54: refcnt=1 paddr=32819
page 55: refcnt=1 paddr=32820
page 56: refcnt=1 paddr=32821
page 57: refcnt=1 paddr=32822
page 58: refcnt=1 paddr=32823
page 59: refcnt=1 paddr=32824
page 60: refcnt=1 paddr=32825
page 61: refcnt=1 paddr=32826
page 62: refcnt=1 paddr=32827
page 63: refcnt=1 paddr=32828
page 64: refcnt=1 paddr=32829
page 65: refcnt=1 paddr=32830
page 66: refcnt=1 paddr=32831
page 67: refcnt=1 paddr=32832
page 68: refcnt=1 paddr=32833
page 69: refcnt=1 paddr=32834
page 70: refcnt=1 paddr=32835
page 71: refcnt=1 paddr=32836
page 72: refcnt=1 paddr=32837
page 73: refcnt=1 paddr=32838
page 74: refcnt=1 paddr=32839
page 75: refcnt=1 paddr=32840
page 76: refcnt=1 paddr=32841
page 77: refcnt=1 paddr=32842
page 78: refcnt=1 paddr=32843
page 79: refcnt=1 paddr=32844
page 80: refcnt=1 paddr=32845
page 81: refcnt=1 paddr=32846
page 82: refcnt=1 paddr=32847
page 83: refcnt=1 paddr=32848
page 84: refcnt=1 paddr=32849
page 85: refcnt=1 paddr=32850
page 86: refcnt=1 paddr=32851
page 87: refcnt=1 paddr=32852
page 88: refcnt=1 paddr=32853
page 89: refcnt=1 paddr=32854
page 90: refcnt=1 paddr=32855
page 91: refcnt=1 paddr=32856
page 92: refcnt=1 paddr=32857
page 93: refcnt=1 paddr=32858
page 94: refcnt=1 paddr=32859
page 95: refcnt=1 paddr=32860
page 96: refcnt=1 paddr=32861
page 97: refcnt=1 paddr=32862
page 98: refcnt=1 paddr=32863
page 99: refcnt=1 paddr=32864
page 100: refcnt=1 paddr=32865
page 101: refcnt=1 paddr=32866
page 102: refcnt=1 paddr=32867
page 103: refcnt=1 paddr=32868
page 104: refcnt=1 paddr=32869
page 105: refcnt=1 paddr=32870
page 106: refcnt=1 paddr=32871
page 107: refcnt=1 paddr=32872
page 108: refcnt=1 paddr=32873
page 109: refcnt=1 paddr=32874
page 110: refcnt=1 paddr=32875
page 111: refcnt=1 paddr=32876
page 112: refcnt=1 paddr=32877
page 113: refcnt=1 paddr=32878
page 114: refcnt=1 paddr=32879
page 115: refcnt=1 paddr=32880
page 116: refcnt=1 paddr=32881
page 117: refcnt=1 paddr=32882
page 118: refcnt=1 paddr=32883
page 119: refcnt=1 paddr=32884
page 120: refcnt=1 paddr=32885
page 121: refcnt=1 paddr=32886
page 122: refcnt=1 paddr=32887
page 123: refcnt=1 paddr=32888
page 124: refcnt=1 paddr=32889
page 125: refcnt=1 paddr=32890
page 126: refcnt=1 paddr=32891
page 127: refcnt=1 paddr=32892
page 128: refcnt=1 paddr=32893
page 129: refcnt=1 paddr=32894
page 130: refcnt=1 paddr=32895
I (1491) main: Found asset partition at offset 0x3f1000 with size c00000
ESP_ERROR_CHECK failed: esp_err_t 0x101 (ESP_ERR_NO_MEM) at 0x42007eb9
0x42007eb9: app_main at D:/esp32-workspace/mmap_test/main/main.c:26 (discriminator 1)

file: "./main/main.c" line 26
func: app_main
expression: esp_partition_mmap(part, 0, part->size, SPI_FLASH_MMAP_DATA, &(mmaped_ptr),&handle)

abort() was called at PC 0x4037a77f on core 0
0x4037a77f: _esp_error_check_failed at C:/Users/ShiroganeTsumugi/esp/esp-idf/components/esp_system/esp_err.c:47



Backtrace: 0x40375eba:0x3fcf3f20 0x4037a789:0x3fcf3f40 0x4037fcee:0x3fcf3f60 0x4037a77f:0x3fcf3fd0 0x42007eb9:0x3fcf4000 0x42019537:0x3fcf4040 0x4037d11d:0x3fcf4070
0x40375eba: panic_abort at C:/Users/ShiroganeTsumugi/esp/esp-idf/components/esp_system/panic.c:412

0x4037a789: esp_system_abort at C:/Users/ShiroganeTsumugi/esp/esp-idf/components/esp_system/esp_system.c:135

0x4037fcee: abort at C:/Users/ShiroganeTsumugi/esp/esp-idf/components/newlib/abort.c:38

0x4037a77f: _esp_error_check_failed at C:/Users/ShiroganeTsumugi/esp/esp-idf/components/esp_system/esp_err.c:47

0x42007eb9: app_main at D:/esp32-workspace/mmap_test/main/main.c:26 (discriminator 1)

0x42019537: main_task at C:/Users/ShiroganeTsumugi/esp/esp-idf/components/freertos/FreeRTOS-Kernel/portable/port_common.c:131 (discriminator 2)

0x4037d11d: vPortTaskWrapper at C:/Users/ShiroganeTsumugi/esp/esp-idf/components/freertos/FreeRTOS-Kernel/portable/xtensa/port.c:151





ELF file SHA256: 497afe5d0adc4f7b

Rebooting...
```

### More Information.

If i turn psram off in menuconfig, the code works fine.
Besides, I tried the code on esp-idf 4.4.2 and it works fine, 
Here's the log on idf 4.4.2
```
I (25) boot: ESP-IDF v4.4.2 2nd stage bootloader
I (25) boot: compile time 00:20:34
I (25) boot: chip revision: 0
I (26) boot.esp32s3: Boot SPI Speed : 80MHz
I (31) boot.esp32s3: SPI Mode       : DIO
I (36) boot.esp32s3: SPI Flash Size : 16MB
I (41) boot: Enabling RNG early entropy source...
I (46) boot: Partition Table:
I (50) boot: ## Label            Usage          Type ST Offset   Length
I (57) boot:  0 otadata          OTA data         01 00 00009000 00002000
I (64) boot:  1 phy_init         RF data          01 01 0000b000 00001000
I (72) boot:  2 factory          factory app      00 00 00010000 00100000
I (79) boot:  3 ota_0            OTA app          00 10 00110000 00100000
I (87) boot:  4 ota_1            OTA app          00 11 00210000 00100000
I (94) boot:  5 nvs              WiFi data        01 02 00310000 000e1000
I (102) boot:  6 asset            unknown          40 00 003f1000 00c00000
I (109) boot: End of partition table
I (114) boot: Defaulting to factory image
I (118) esp_image: segment 0: paddr=00010020 vaddr=3c020020 size=07e20h ( 32288) map
I (133) esp_image: segment 1: paddr=00017e48 vaddr=3fc90fb0 size=029d8h ( 10712) load
I (137) esp_image: segment 2: paddr=0001a828 vaddr=40374000 size=057f0h ( 22512) load
I (149) esp_image: segment 3: paddr=00020020 vaddr=42000020 size=173f4h ( 95220) map
I (169) esp_image: segment 4: paddr=0003741c vaddr=403797f0 size=077c0h ( 30656) load
I (176) esp_image: segment 5: paddr=0003ebe4 vaddr=50000000 size=00010h (    16) load
I (182) boot: Loaded app from partition at offset 0x10000
I (182) boot: Disabling RNG early entropy source...
I (198) opi psram: vendor id : 0x0d (AP)
I (198) opi psram: dev id    : 0x02 (generation 3)
I (198) opi psram: density   : 0x03 (64 Mbit)
I (202) opi psram: good-die  : 0x01 (Pass)
I (207) opi psram: Latency   : 0x01 (Fixed)
I (211) opi psram: VCC       : 0x01 (3V)
I (216) opi psram: SRF       : 0x01 (Fast Refresh)
I (222) opi psram: BurstType : 0x01 (Hybrid Wrap)
I (227) opi psram: BurstLen  : 0x01 (32 Byte)
I (232) opi psram: Readlatency  : 0x02 (10 cycles@Fixed)
I (238) opi psram: DriveStrength: 0x00 (1/1)
I (243) spiram: Found 64MBit SPI RAM device
I (248) spiram: SPI RAM mode: sram 40m
I (252) spiram: PSRAM initialized, cache is in normal (1-core) mode.
I (259) cpu_start: Pro cpu up.
I (263) cpu_start: Starting app cpu, entry point is 0x403751c4
0x403751c4: call_start_cpu1 at D:/idf4.4/components/esp_system/port/cpu_start.c:160

I (0) cpu_start: App cpu up.
I (1000) spiram: SPI SRAM memory test OK
I (1009) cpu_start: Pro cpu start user code
I (1010) cpu_start: cpu freq: 160000000
I (1010) cpu_start: Application information:
I (1013) cpu_start: Project name:     template-app
I (1018) cpu_start: App version:      1
I (1023) cpu_start: Compile time:     Sep 24 2022 00:19:55
I (1029) cpu_start: ELF file SHA256:  e0f8f45297fb3197...
I (1035) cpu_start: ESP-IDF:          v4.4.2
I (1040) heap_init: Initializing. RAM available for dynamic allocation:
I (1047) heap_init: At 3FC94380 len 0004BC80 (303 KiB): D/IRAM
I (1054) heap_init: At 3FCE0000 len 0000EE34 (59 KiB): STACK/DRAM
I (1060) heap_init: At 3FCF0000 len 00008000 (32 KiB): DRAM
I (1067) heap_init: At 600FE000 len 00002000 (8 KiB): RTCRAM
I (1073) spiram: Adding pool of 8192K of external SPI memory to heap allocator
I (1081) spi_flash: detected chip: gd
I (1085) spi_flash: flash io: dio
I (1090) sleep: Configure to isolate all GPIO pins in sleep state
I (1096) sleep: Enable automatic switching of GPIO sleep configuration
I (1104) cpu_start: Starting scheduler on PRO CPU.
I (0) cpu_start: Starting scheduler on APP CPU.
I (1124) spiram: Reserving pool of 32K of internal memory for DMA/internal allocations
page 0: refcnt=1 paddr=2
page 1: refcnt=1 paddr=3
page 2: refcnt=1 paddr=1
page 384: refcnt=1 paddr=32768
page 385: refcnt=1 paddr=32769
page 386: refcnt=1 paddr=32770
page 387: refcnt=1 paddr=32771
page 388: refcnt=1 paddr=32772
page 389: refcnt=1 paddr=32773
page 390: refcnt=1 paddr=32774
page 391: refcnt=1 paddr=32775
page 392: refcnt=1 paddr=32776
page 393: refcnt=1 paddr=32777
page 394: refcnt=1 paddr=32778
page 395: refcnt=1 paddr=32779
page 396: refcnt=1 paddr=32780
page 397: refcnt=1 paddr=32781
page 398: refcnt=1 paddr=32782
page 399: refcnt=1 paddr=32783
page 400: refcnt=1 paddr=32784
page 401: refcnt=1 paddr=32785
page 402: refcnt=1 paddr=32786
page 403: refcnt=1 paddr=32787
page 404: refcnt=1 paddr=32788
page 405: refcnt=1 paddr=32789
page 406: refcnt=1 paddr=32790
page 407: refcnt=1 paddr=32791
page 408: refcnt=1 paddr=32792
page 409: refcnt=1 paddr=32793
page 410: refcnt=1 paddr=32794
page 411: refcnt=1 paddr=32795
page 412: refcnt=1 paddr=32796
page 413: refcnt=1 paddr=32797
page 414: refcnt=1 paddr=32798
page 415: refcnt=1 paddr=32799
page 416: refcnt=1 paddr=32800
page 417: refcnt=1 paddr=32801
page 418: refcnt=1 paddr=32802
page 419: refcnt=1 paddr=32803
page 420: refcnt=1 paddr=32804
page 421: refcnt=1 paddr=32805
page 422: refcnt=1 paddr=32806
page 423: refcnt=1 paddr=32807
page 424: refcnt=1 paddr=32808
page 425: refcnt=1 paddr=32809
page 426: refcnt=1 paddr=32810
page 427: refcnt=1 paddr=32811
page 428: refcnt=1 paddr=32812
page 429: refcnt=1 paddr=32813
page 430: refcnt=1 paddr=32814
page 431: refcnt=1 paddr=32815
page 432: refcnt=1 paddr=32816
page 433: refcnt=1 paddr=32817
page 434: refcnt=1 paddr=32818
page 435: refcnt=1 paddr=32819
page 436: refcnt=1 paddr=32820
page 437: refcnt=1 paddr=32821
page 438: refcnt=1 paddr=32822
page 439: refcnt=1 paddr=32823
page 440: refcnt=1 paddr=32824
page 441: refcnt=1 paddr=32825
page 442: refcnt=1 paddr=32826
page 443: refcnt=1 paddr=32827
page 444: refcnt=1 paddr=32828
page 445: refcnt=1 paddr=32829
page 446: refcnt=1 paddr=32830
page 447: refcnt=1 paddr=32831
page 448: refcnt=1 paddr=32832
page 449: refcnt=1 paddr=32833
page 450: refcnt=1 paddr=32834
page 451: refcnt=1 paddr=32835
page 452: refcnt=1 paddr=32836
page 453: refcnt=1 paddr=32837
page 454: refcnt=1 paddr=32838
page 455: refcnt=1 paddr=32839
page 456: refcnt=1 paddr=32840
page 457: refcnt=1 paddr=32841
page 458: refcnt=1 paddr=32842
page 459: refcnt=1 paddr=32843
page 460: refcnt=1 paddr=32844
page 461: refcnt=1 paddr=32845
page 462: refcnt=1 paddr=32846
page 463: refcnt=1 paddr=32847
page 464: refcnt=1 paddr=32848
page 465: refcnt=1 paddr=32849
page 466: refcnt=1 paddr=32850
page 467: refcnt=1 paddr=32851
page 468: refcnt=1 paddr=32852
page 469: refcnt=1 paddr=32853
page 470: refcnt=1 paddr=32854
page 471: refcnt=1 paddr=32855
page 472: refcnt=1 paddr=32856
page 473: refcnt=1 paddr=32857
page 474: refcnt=1 paddr=32858
page 475: refcnt=1 paddr=32859
page 476: refcnt=1 paddr=32860
page 477: refcnt=1 paddr=32861
page 478: refcnt=1 paddr=32862
page 479: refcnt=1 paddr=32863
page 480: refcnt=1 paddr=32864
page 481: refcnt=1 paddr=32865
page 482: refcnt=1 paddr=32866
page 483: refcnt=1 paddr=32867
page 484: refcnt=1 paddr=32868
page 485: refcnt=1 paddr=32869
page 486: refcnt=1 paddr=32870
page 487: refcnt=1 paddr=32871
page 488: refcnt=1 paddr=32872
page 489: refcnt=1 paddr=32873
page 490: refcnt=1 paddr=32874
page 491: refcnt=1 paddr=32875
page 492: refcnt=1 paddr=32876
page 493: refcnt=1 paddr=32877
page 494: refcnt=1 paddr=32878
page 495: refcnt=1 paddr=32879
page 496: refcnt=1 paddr=32880
page 497: refcnt=1 paddr=32881
page 498: refcnt=1 paddr=32882
page 499: refcnt=1 paddr=32883
page 500: refcnt=1 paddr=32884
page 501: refcnt=1 paddr=32885
page 502: refcnt=1 paddr=32886
page 503: refcnt=1 paddr=32887
page 504: refcnt=1 paddr=32888
page 505: refcnt=1 paddr=32889
page 506: refcnt=1 paddr=32890
page 507: refcnt=1 paddr=32891
page 508: refcnt=1 paddr=32892
page 509: refcnt=1 paddr=32893
page 510: refcnt=1 paddr=32894
page 511: refcnt=1 paddr=32895
I (1494) main: Found asset partition at offset 0x3f1000 with size c00000
I (1504) main: Mounted at offset 0x3c031000
hello world!
hello world!
hello world!
hello world!
```

**Comment by igrr** [comment]
It looks like a limitation of the available address space. ESP32-S3 has 32MB of address space available for external Flash and Psram. 8MB is occupied by PSRAM, some space is occupied by the app itself, which leaves less than 12 MB available, so the attempt to mount fails.

**Comment by AxelLin** [comment]
> Edit: after looking at the the dump output (sorry that i didn't do this in the first place) there should be enough pages available (192). So this might indeed be a bug!

How is the status of this issue now?

**Comment by inapp123** [comment]
> > Edit: after looking at the the dump output (sorry that i didn't do this in the first place) there should be enough pages available (192). So this might indeed be a bug!
> 
> How is the status of this issue now?

With the latest ESP-IDF stable release(5.1.2), it just runs flawlessly...
It might be fixed around 5.0.x, according to toitlang/toit#1555
